pkg/sync/syncer.go

```diff
@@ -761,15 +772,15 @@ func WithTransitionHandler(f func(s Action)) SyncOpt {
 }
 
 // NewSyncer returns a new syncer object.
-func NewSyncer(store connectorstore.Writer, c types.ClientWrapper, opts ...SyncOpt) Syncer {
+func NewSyncer(ctx context.Context, c types.ConnectorClient, dbPath string, opts ...SyncOpt) (Syncer, error) {
```

Comment:
Need to think more about this prototype and make sure it makes sense for SaaS use cases. I don't think this is actually what we want today.